### PR TITLE
Update CICE

### LINKED ISSUE #61: Update CICE to latest Consortium/main
Update NOAA-EMC fork for updates from Consortium

- Deprecation of some features https://github.com/CICE-Consortium/CICE/pull/841
- Fix time axis documentation https://github.com/CICE-Consortium/CICE/pull/839
- Modify mesh check https://github.com/CICE-Consortium/CICE/pull/830
- Improve global sum efficiency https://github.com/CICE-Consortium/CICE/pull/824
- Adds https://github.com/CICE-Consortium/CICE/pull/833

### LINKED ISSUE #1801: Remove un-used "history" settings in various nems.configure templates
## Description

Many of the ``nems.configure`` templates contain the settings

```
      history_n = 1
      history_option = nhours
      history_ymd = -999
```

In almost all cases however, these are un-unsed since the UFS a) does not compare mediator history files in the RTs and b) the run sequences do not in general contain the run phase ``MED med_phases_history_write`` which uses the above setting.

Also, the mediator history functionality which allows for history files for individual components (see Discussion https://github.com/ufs-community/ufs-weather-model/discussions/934) has in most cases superceded the use of the special history write phase in any case.

## Solution

Remove the un-unused settings in the nems.configure templates.

### LINKED ISSUE #1826: Set diagnostic frequency for CICE to a reasonable value
## Description

The ice_in value of ``diag_freq`` controls the printing of CICE diagnostic information, for example hemispheric ice volumes and area.

This value is the frequency in timesteps, not hours. Currently in RTs, it is set as ``6`` in all tests. For the mx025 cases, this means the diagnostic calculation occurs 48/day. This is much too frequent, especially because no information is extracted from the ``ice_diag.d`` file. 

## Solution

Create a variable for setting the ``diag_freq`` according to the relevant ``dt`` such that it is written one time only (e.g. once/day for 24 hour tests, or once/3 hours for the debug c384mx024 test).

### PR DESCRIPTION
**Pull request by DeniseWorthen**
## Description

Update CICE to latest Consortium/main

### Input data additions/changes
- [x] No changes are expected to input data.
- [ ] Changes are expected to input data:
  - [ ] New input data.
  - [ ] Updated input data.

### Anticipated changes to regression tests:
- [x] No changes are expected to any regression test.
- [ ] Changes are expected to the following tests:

[RegressionTests_hera.log](https://github.com/ufs-community/ufs-weather-model/files/12165894/RegressionTests_hera.log)

## Subcomponents involved:
- [ ] AQM
- [ ] CDEPS
- [x] CICE
- [ ] CMEPS
- [ ] CMakeModules
- [ ] FV3
- [ ] GOCART
- [ ] HYCOM
- [ ] MOM6
- [ ] NOAHMP
- [ ] WW3
- [ ] stochastic_physics
- [ ] none

### Library Updates/Changes

- [X] Not Needed
- [ ] Create separate issue in [JCSDA/spack-stack](https://github.com/JCSDA/spack-stack) asking for update to library. Include library name, library version.
- [ ] Add issue link from JCSDA/spack-stack following this item

## Commit Queue Checklist:

- [x] Link PR's from all sub-components involved in section below
- [x] Confirm reviews completed in ALL sub-component PR's
- [x] Add all appropriate labels to this PR.
- [x] Run full RT suite on either Hera/Cheyenne AND attach log to a PR comment.
- [x] Add list of any failed regression tests to "Anticipated changes to regression tests" section.

## Linked PR's and Issues:

- Fixes https://github.com/ufs-community/ufs-weather-model/issues/1826
- Fixes https://github.com/ufs-community/ufs-weather-model/issues/1801
- Fixes https://github.com/NOAA-EMC/CICE/issues/61
- Depends on https://github.com/NOAA-EMC/CICE/pull/62

## Testing Day Checklist:

- [x] This PR is up-to-date with the top of all sub-component repositories except for those sub-components which are the subject of this PR.
- [ ] Move new/updated input data on RDHPCS Hera and propagate input data changes to all supported systems.

### Testing Log (for CM's):
- RDHPCS
  - [x] Hera
  - [x] Orion
  - [x] Jet
  - [x] Gaea
  - [ ] Cheyenne N/A (offline)
- WCOSS2
  - [x] Dogwood/Cactus
  - [ ] Acorn
- CI
  - [x] Completed
- opnReqTest
  - [ ] N/A
  - [x] Log attached to comment

### CODE REVIEW

**Comment by sadeghitabas** [comment]
@DeniseWorthen Could you please update the gitmodules? Thanks

**Comment by DeniseWorthen** [comment]
@SadeghTabas-NOAA I don't understand. The gitmodules points to the feature branch for CICE.

**Comment by sadeghitabas** [comment]
> @SadeghTabas-NOAA I don't understand. The gitmodules points to the feature branch for CICE.

Oh, sorry I was wrong.

**Review by sadeghitabas — DISMISSED** [review]

**Comment by FernandoAndrade-NOAA** [comment]
Ok, successful runs were verified on several machines for issue #1847. @BrianCurtis-NOAA getting started with jenkins ci for this PR

**Comment by FernandoAndrade-NOAA** [comment]
Jenkins ci logs are attached, ORTs passed.
[ufs-weather-model » ort-docker-pipeline » PR-1842 #1 Console [Jenkins].txt](https://github.com/ufs-community/ufs-weather-model/files/12234345/ufs-weather-model.ort-docker-pipeline.PR-1842.1.Console.Jenkins.txt)

File: tests/parm/nems.configure.atmw.IN [hunk 1]
```diff
@@ -68,9 +68,6 @@ DRIVER_attributes::
 MED_attributes::
       ATM_model = @[atm_model]
       WAV_model = @[wav_model]
-      history_n = 1
```
Comment (junwang-noaa) [inline tests/parm/nems.configure.atmw.IN:71]: If we remove this line, what default value history_n has?

Comment (DeniseWorthen) [reply tests/parm/nems.configure.atmw.IN:71]: This is an optional configuration that is behind an if present, if set logical. So it usually defaults to -999 if not set.

This way of specifying the writing of optional history files has been super-ceded by the ability to write component history files. The older style requires the addition of a mediator history write phase in the run sequence and all components write into the same file. These settings do nothing w/o that history write phase being present in the run sequence.

The newer functionality does not require any change to the run sequence---mediator history files can be turned on/off with configuration settings and each component can output multiple types of history files---instantaneous, time-averaged etc.

**Review by junwang-noaa — DISMISSED** [review]

**Review by sadeghitabas — DISMISSED** [review]

**Comment by DeniseWorthen** [comment]
@BrianCurtis-NOAA Any hope of getting an Acorn log or is it flakey today?

**Comment by BrianCurtis-NOAA** [comment]
@FernandoAndrade-NOAA @DeniseWorthen The full disk earlier seemed to mess with my baselines from the previous PR, so i'll re-do them tonight to confirm, with no baseline change, shouldn't be a problem to skip to merge now.

**Comment by FernandoAndrade-NOAA** [comment]
> @FernandoAndrade-NOAA @DeniseWorthen The full disk earlier seemed to mess with my baselines from the previous PR, so i'll re-do them tonight to confirm, with no baseline change, shouldn't be a problem to skip to merge now.

Understood, in that case this PR should be ready to begin the merge process.

File: tests/parm/nems.configure.atmw.IN (hunk 1 above)
Comment (FernandoAndrade-NOAA) [reply tests/parm/nems.configure.atmw.IN:71]: @junwang-noaa, would you consider Denise's answer to be a resolution to this discussion / review?

**Review by jkbk2004 — APPROVED** [review]

**Review by sadeghitabas — APPROVED** [review]

File: tests/parm/nems.configure.atmw.IN (hunk 1 above)
Comment (FernandoAndrade-NOAA) [reply tests/parm/nems.configure.atmw.IN:71]: I believe this conversation is resolved, continuing with merge process.

**Pull request merged**